### PR TITLE
Replacing min_time with reachable distance calculation

### LINKED ISSUE #3609: Enhanced sniping crash
#3277 added new config options for sniping capability
- "snipe_high_prio_only": true,
- "snipe_high_prio_threshold": 400,

### Expected Behavior
Anyone on a commit before this merge can use the bot, without adding the new config options as the code should default values for users who don't add the new config options

### Actual Behavior
Bot crashes as the config options do not default in the code

### Steps to Reproduce
git dev branch sitting at pre #3277 
git pull
python pokecli.py
Crashes when trying to snipe
http://pastebin.com/LTYCn5eQ

### Other Information
OS: Windows 8.1 x64
Git Commit: 0292c2f426b2432be70b0ca867fa62d8436cb5bf
Python Version: 2.7.12

### PR DESCRIPTION
**Pull request by Raiven66**
## Short Description:

* Replacing min_time functionality of MoveToMapPokemon by a more sophisticated way of determining reachability on time. This way it can be more reliably decided whether to run after a pokemon or not.

## Fixed Issues

* Closes #3609

### CODE REVIEW

**Comment by Raiven66** [comment]
found a bug with sniping, closing

**Comment by douglascamata** [comment]
👍 

@TheSavior can you review this, please?

File: pokemongo_bot/cell_workers/move_to_map_pokemon.py [hunk 1]
```diff
@@ -257,8 +259,8 @@ def work(self):
             return WorkerResult.SUCCESS
 
         if self.config['snipe']:
-            if self.config['snipe_high_prio_only']:
-                if self.config['snipe_high_prio_threshold'] < pokemon['priority'] or pokemon['is_vip']:
+            if self.config.get('snipe_high_prio_only', False):
+                if self.config.get('snipe_high_prio_threshold', 400) < pokemon['priority'] or pokemon['is_vip']:
```
Comment (TheSavior) [inline pokemongo_bot/cell_workers/move_to_map_pokemon.py]: This all looks great except these. These defaults should go up after line 87 with the other config defaults for this file.

Comment (Raiven66) [inline pokemongo_bot/cell_workers/move_to_map_pokemon.py]: Will move them, should I move the 259 `self.config['snipe']` there as well?

**Comment by TheSavior** [comment]
:+1:

**Comment by NovaDev94** [comment]
@Raiven66 Please fix https://github.com/PokemonGoF/PokemonGo-Bot/commit/e6c8f5303f3d8b63ef3ca6297eb560797399b2ff#diff-13dfb7470ca1360a74ad4eabf7bb11a2R147

from `pokemon['expire']` => `pokemon['disappear_time']`. I'm having an exception 

> 2016-08-17 03:05:04,386 [sentry.errors.uncaught] [ERROR] [u"KeyError: 'expire'", u'  File "pokecli.py", line 626, in <module>', u'  File "pokecli.py", line 104, in main', u'  File "pokemongo_bot/__init__.py", line 498, in tick', u'  File "pokemongo_bot/cell_workers/move_to_map_pokemon.py", line 248, in work', u'  File "pokemongo_bot/cell_workers/move_to_map_pokemon.py", line 147, in get_pokemon_from_map']

My MoveToMapPokemon config. And I'm having a `PokemonGo-Map` instance running on `localhost:5000`.  
```json
            "type": "MoveToMapPokemon",
            "config": {
                "address": "http://localhost:5000",
                "max_distance": 500,
                "min_ball": 75,
                "prioritize_vips": true,
                "snipe": true,
                "snipe_high_prio_only": true,
                "snipe_high_prio_threshold": 400,
                "update_map": true,
                "mode": "priority",
                "map_path": "raw_data",
                "catch": {}
            }
```

**Comment by Raiven66** [comment]
will do, thanks

**Pull request merged**